src/config.ts

```diff
@@ -23,14 +24,23 @@ export interface LoadGeneratorConfiguration {
 
 export const readConfig = (): ActionConfig => {
   const gatlingEnterpriseUrl = getGatlingEnterpriseUrlConfig();
-  const config = { gatlingEnterpriseUrl, api: getApiConfig(gatlingEnterpriseUrl), run: getRunConfig() };
+  const failActionOnRunFailure = getFailActionOnRunFailureConfig();
+  const config = {
+    gatlingEnterpriseUrl,
+    api: getApiConfig(gatlingEnterpriseUrl),
+    run: getRunConfig(),
+    failActionOnRunFailure
+  };
   logDebug("Parsed configuration: " + JSON.stringify({ api: { ...config.api, apiToken: "*****" }, run: config.run }));
   return config;
 };
 
 const getGatlingEnterpriseUrlConfig = (): string =>
   getValidatedInput("gatling_enterprise_url", requiredInputValidation, "gatling_enterprise_url is required");
 
+const getFailActionOnRunFailureConfig = (): boolean =>
+  getValidatedInput("fail_action_on_run_failure", requiredBooleanValidation, "fail_on_run_failure is required");
```

Comment:
In `action.yml`:

>     required: false

but here, the code warns that it's required?

I'm confused

Reply:
In the action definition: it's not required because there is a default value.

In the code: it's required, though in practice it will never be missing when actually running the code as an action, since the default value will be provided by the runner.